crates/conjure_core/src/rules/utils.rs

```diff
@@ -36,6 +37,33 @@ pub fn is_all_constant(expression: &Expr) -> bool {
     true
 }
 
+/// Creates a single `Expr` from a `Vec<Expr>` by forming the conjunction.
+///
+/// If it contains a single element, that element is returned, otherwise the conjunction of the
+/// elements is returned.
+///
+/// # Returns
+///
+/// - Some: the expression list is non empty, and all expressions are booleans (so can be
+/// conjuncted).
+///
+/// - None: the expression list is empty, or not all expressions are booleans.
+pub fn exprs_to_conjunction(exprs: &Vec<Expr>) -> Option<Expr> {
+    match exprs.as_slice() {
+        [] => None,
+        [a] if a.return_type() == Some(ReturnType::Bool) => Some(a.clone()),
+        [_, _, ..] => {
+            for expr in exprs {
+                if expr.return_type() != Some(ReturnType::Bool) {
+                    return None;
+                }
+            }
+            Some(Expr::And(Metadata::new(), exprs.clone()))
+        }
+        _ => None,
```

Comment:
this case should be redundant? why not change line 55 to be _ and remove line 63?